checkpoint.go

```diff
@@ -36,7 +36,8 @@ type CheckParams struct {
 	Version string
 
 	// Generic product flags
-	Flags map[string]string
+	Flags      map[string]string
+	ExtraFlags func() []Flag
```

Comment:
not breaking backwards compatibility; helping people who don't want to create a function to keep some static data around